homeassistant/components/smart_meter_texas/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for Smart Meter Texas integration."""
+import asyncio
+import logging
+
+from aiohttp import ClientError
+from smart_meter_texas.async_api import Auth, SMTError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required("meter"): str,
+        vol.Required("esiid"): str,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    client_session = aiohttp_client.async_get_clientsession(hass)
+
+    try:
+        auth = Auth(client_session, data["username"], data["password"])
+        await auth.authenticate()
+    except (asyncio.TimeoutError, ClientError):
+        raise CannotConnect
+    except SMTError:
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": "Electric Meter"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Smart Meter Texas."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
```

Comment:
```
    def _host_already_configured(self, host):
        """See if we already have a hub with the host address configured."""
        existing_hosts = {
            entry.data[CONF_HOST]
            for entry in self._async_current_entries()
            if CONF_HOST in entry.data
        }
        return host in existing_hosts
```
```
            if self._host_already_configured(user_input[CONF_HOST]):
                return self.async_abort(reason="already_configured")
```
We should do something like this (except for the username) to make sure they don't create duplicate config entries.